### DISCUSSION TITLE
How/Where to use websocket (binance-connector-typescript)

### DISCUSSION THREAD

**Discussion by JohnPAfr** [nuxt/nuxt#30075] [Questions]
Hello, I'm very new to nuxt and backend in general. I'm working on an app to retrieve data from Binance.

They provide a Restfull API which I have no particular issue using for simple usecases.

They also provide Websockets which could be what I really need but two problems:
1. I have zero knowledge on ws (so yeah maybe learn more about that)
2. Where should I use it on a nuxt3 app.

(https://github.com/binance/binance-connector-typescript)

I asked ChatGPT and Claude for some help.
Chat doesn't use the correct library and doesn't really help me on the "architecture".

Claude provides a bit more but not sure if I can follow what he shares.
One solution was to make a composable and use it in a component. But it felt weird to connect to a ws in a component.
The others was to make a nuxt plugin.

I'm clearly lost and looking for some guidance :)

(maybe I should not use Nuxt for that)

**Comment by ndragun92** [comment]
You can use `socket.io` same as you would use in any other application/framework.

You can create a plugin in nuxt 3 (as client plugin)
and then subscribe to things you want to listen and do your logic.

You can also use their `import { WebsocketAPI } from '@binance/connector-typescript';`

And create a plugin. Should be pretty simple. Nuxt wont prevent you doing stuff you need.

Only thing you have to worry is in case something is not meant to be server side rendered you just have to load it on client which is a pretty easy thing to do in multiple ways.

@JohnPAfr
